### PR TITLE
pythonPackages.pytorch-lightning: init at 0.5.3.2

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change
Add PyTorch Lightning, the lightweight PyTorch wrapper for ML researchers

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jonringer** [comment]
unfortunately, pytorch is broken right now, so this doesn't build

**Comment by tbenst** [comment]
No worries, can wait until the major python packages update is merged. It does build on my pull request branch (which is an old commit on master from Dec 12). Also, with [cherry-pick patches](https://github.com/tbenst/nixpkgs/commits/nix-data) we have this [building on latest nixpkgs unstable](https://hydra.nix-data.org/build/109).

File: pkgs/development/python-modules/pytorch-lightning/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pytorch-lightning/default.nix]: please add yourself as a maintainer

File: pkgs/development/python-modules/pytorch-lightning/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pytorch-lightning/default.nix]: you don't have to run all the tests
```suggestion
  py.test pytorch_lightning tests -k 'not this_network_test'
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

failures are from python38Packages.torch
```
[6 built (1 failed), 3 copied (9.1 MiB), 3.6 MiB DL]
error: build of '/nix/store/43mvmb2wnaqii07q1aqyz47jhxha43a4-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/77155
2 package failed to build:
python38Packages.pytorch-lightning python38Packages.test-tube

2 package built:
python37Packages.pytorch-lightning python37Packages.test-tube
```

**Comment by jonringer** [comment]
please squash fix-ups

**Comment by tbenst** [comment]
Squashed, building overnight. Likely need to exclude additional tests that require internet

**Review by doronbehar — CHANGES_REQUESTED** [review]
nixpkgs-review fails with:

```
builder for '/nix/store/hpb8g409pf92l2zg1hwz6i9bkzin73r7-python3.7-pytorch-lightning-0.5.3.2.drv' failed with exit code 4; last 10 log lines:
  running install tests
  ============================= test session starts ==============================
  platform linux -- Python 3.7.7, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /nix/store/2dcsn57cgaxs92ha5swihrab0g3l2h6g-python3-3.7.7/bin/python3.7
  cachedir: .pytest_cache
  rootdir: /build/source, inifile: setup.cfg
  collected 0 items                                                              
  
  ============================ no tests ran in 0.00s =============================
  ERROR: file not found: test_running_test_pretrained_model
  
builder for '/nix/store/yray8z3h8lw93zsb5fb3c330zbrcqqzv-python3.8-pytorch-lightning-0.5.3.2.drv' failed with exit code 4; last 10 log lines:
  running install tests
  ============================= test session starts ==============================
  platform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /nix/store/081w0sp7jnh33548cihj3a9y9jpn9b3s-python3-3.8.2/bin/python3.8
  cachedir: .pytest_cache
  rootdir: /build/source, inifile: setup.cfg
  collected 0 items                                                              
  
  ============================ no tests ran in 0.00s =============================
  ERROR: file not found: test_running_test_pretrained_model
  
cannot build derivation '/nix/store/6ybfn8l8xfcpdbpp7xi57yn6hrlybpk4-env.drv': 2 dependencies couldn't be built
[4 built (2 failed), 65 copied (1083.5 MiB), 202.0 MiB DL]
error: build of '/nix/store/6ybfn8l8xfcpdbpp7xi57yn6hrlybpk4-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/77155
2 packages failed to build:
python37Packages.pytorch-lightning python38Packages.pytorch-lightning

2 packages built:
python37Packages.test-tube python38Packages.test-tube
```

**Comment by bcdarwin** [comment]
Hi @tbenst -- any chance of updating this?  If not I'll make a new PR.

**Comment by tbenst** [comment]
Hi @bcdarwin, would welcome the help! Please feel free to push to this PR or make a new PR, I believe the package is fully functional, and I've used it, but getting the tests running will require downloading pre-trained models & data ahead of time, and possibly patching the test code to avoid a download. If you want to chat realtime about this, feel free to ping me on the [nix-data slack channel](https://join.slack.com/t/nix-data/shared_invite/zt-ca8csgcz-N9Fyh~tnoZPY8x5lE_slFA).

File: pkgs/development/python-modules/test-tube/default.nix
Comment (bcdarwin) [inline pkgs/development/python-modules/test-tube/default.nix:43]: `maintainers` should be set

File: pkgs/development/python-modules/test-tube/default.nix
Comment (bcdarwin) [inline pkgs/development/python-modules/test-tube/default.nix:41]: homepage URLs should be quoted now (in both packages)

**Review by bcdarwin — CHANGES_REQUESTED** [review]

**Comment by bcdarwin** [comment]
I added a couple cosmetic comments inline, but I guess the main thing is updating the package (0.7.6 is out) so that it can work with Pytorch 1.5.

**Comment by tbenst** [comment]
@bcdarwin, yes, updating will require adding several packages to nixpkgs: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/environment.yml. In particular, trains, Neptune, wandb, and comet ML, although as I did in 0.5.3.2 we can potentially remove imports that are unused (I maintain mlflow but don’t have interest in the others). But than you for raising this point, doesn’t make sense to add tests right now.

In fact, I might propose that we skip tests on this package for now if others are ok with it, as the maintenance burden is extraordinarily high given the rate of releases, change in testing environment, and high utilization of network to download pre-trained models and datasets

**Comment by bcdarwin** [comment]
Closing in favour of #96119.

**Pull request closed without merging**
